File: src/main/java/lechuck/intellij/TaskConfigurationFactory.java
package lechuck.intellij;

import com.intellij.execution.configurations.ConfigurationFactory;
import com.intellij.execution.configurations.ConfigurationType;
import com.intellij.execution.configurations.RunConfiguration;
import com.intellij.openapi.components.BaseState;
import com.intellij.openapi.project.Project;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class TaskConfigurationFactory extends ConfigurationFactory {

    public TaskConfigurationFactory(ConfigurationType type) {
        super(type);
    }

    @Override
    public @NotNull @NonNls String getId() {
        return TaskRunConfigurationType.ID;
    }

    @Override
    public @NotNull RunConfiguration createTemplateConfiguration(@NotNull Project project) {
        return new TaskRunConfiguration(project, this, "Task");
    }

    @Override
    public @Nullable Class<? extends BaseState> getOptionsClass() {
        return TaskRunConfigurationOptions.class;
    }
}

File: src/main/kotlin/lechuck/intellij/TaskConfigurationFactory.kt
package lechuck.intellij

import com.intellij.execution.configurations.ConfigurationFactory
import com.intellij.openapi.project.Project

class TaskConfigurationFactory(private val runCfgType: TaskRunConfigurationType) : ConfigurationFactory(runCfgType) {

    override fun getId() = TaskRunConfigurationType.ID

    override fun getName(): String = runCfgType.displayName

    override fun createTemplateConfiguration(project: Project) = TaskRunConfiguration(project, this, "Task")
}